internal/provider/resource_gitlab_runner.go

```diff
@@ -0,0 +1,257 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_runner", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`gitlab_runner`" + ` resource allows registering a runner, either at an instance level
+or at a group level. The runner will be registered at a group level if the token used is from a group, or at an 
+instance level if the token used is for the instance.
+
+**Upstream API**: [GitLab REST API docs](https://docs.gitlab.com/ee/api/runners.html#register-a-new-runner)`,
+
+		// Since this resource updates an in-place resource, the update method is the same as the create method
+		CreateContext: resourceGitLabRunnerCreate,
+		UpdateContext: resourceGitLabRunnerUpdate,
+		ReadContext:   resourceGitLabRunnerRead,
+		DeleteContext: resourceGitLabRunnerDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"token": {
+				Description: `The registration token used to register the runner`,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Sensitive:   true,
+			},
+			// Even though the output variable is just called 'token', we need to differentiate between the regitration
+			// and authentication token in terraform.
+			"authentication_token": {
+				Description: `The authentication token used for building a config.toml file. This value is not present when imported.`,
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"status": {
+				Description: `The status of runners to show, one of: online and offline. active and paused are also possible values
+				 which were deprecated in GitLab 14.8 and will be removed in GitLab 16.0`,
```

Comment:
This indentation yields to a strange formatting (see the md) ...

Reply:
@timofurrer - true, but the markdown still renders correctly. Do you think we want to fix it? Since the documentation is generated, I'd personally favor the readability of the code in the func as opposed to the indentation of the markdown file:

![image](https://user-images.githubusercontent.com/50960557/166168537-68f8307c-afc3-4ab5-9740-48c384f78431.png)

Reply:
Okay, but the alignment to me is still strange in the code with the one space indentation.
I'd rather align on the `The status ...` then ..

Reply:
Oh, sure. That's easy enough to fix!

Reply:
Updated in last commit, just to be explicit 😄